nautobot/utilities/renderers.py

```diff
@@ -1,15 +1,169 @@
+                form = view._import_form_for_bulk_import()
+            elif view.action == "bulk_edit":
+                if request.POST.get("_all") and view.bulk_edit_filterset is not None:
+                    pk_list = [obj.pk for obj in view.bulk_edit_filterset(request.GET, view.queryset.only("pk")).qs]
+                else:
+                    pk_list = request.POST.getlist("pk")
+                if "_apply" in request.POST:
+                    form = view.bulk_edit_form(model, request.POST)
+                    restrict_form_fields(form, request.user)
+                else:
+                    # Include the PK list as initial data for the form
+                    initial_data = {"pk": pk_list}
+                    # Check for other contextual data needed for the form. We avoid passing all of request.GET because the
+                    # filter values will conflict with the bulk edit form fields.
+                    # TODO: Find a better way to accomplish this
+                    if "device" in request.GET:
+                        initial_data["device"] = request.GET.get("device")
+                    elif "device_type" in request.GET:
+                        initial_data["device_type"] = request.GET.get("device_type")
+
+                    form = view.bulk_edit_form(model, initial=initial_data)
+                    restrict_form_fields(form, request.user)
+
+        context = {
+            "obj": obj,
+            "object": instance,
+            "obj_type": view.queryset.model._meta.verbose_name,
+            "obj_type_plural": view.queryset.model._meta.verbose_name_plural,
+            "editing": obj.present_in_database,
+            "form": form,
+            "fields": view.import_form().fields,
+            "table": table if table else data.get("table", None),
+            "return_url": view.get_return_url(request, instance),
+            "verbose_name": view.queryset.model._meta.verbose_name,
+            "verbose_name_plural": view.queryset.model._meta.verbose_name_plural,
+            "changelog_url": view.get_changelog_url(instance),
+            "content_type": content_type,
+            "permissions": permissions,
+            "action_buttons": valid_actions,
+            "table_config_form": TableConfigForm(table=table) if table else None,
+            "filter_form": view.filterset_form(request.GET, label_suffix="") if view.filterset_form else None,
+            "active_tab": "csv-data",
+            **detail_context,
+        }
+        context.update(view.get_extra_context(request, ACTION_TO_VIEW_TYPE[view.action], instance=None))
         return context
 
     def render(self, data, accepted_media_type=None, renderer_context=None):
-        self.template = data["template"]
+        view = renderer_context["view"]
+        if data.get("template"):
+            self.template = data["template"]
+        else:
+            self.template = view.get_template_name(ACTION_TO_VIEW_TYPE[view.action])
```

Comment:
I don't think this is correct, since we are moving the template names to align with the DRF view actions and not the translated (legacy) Nautobot view actions that correspond to route names. I believe that it should be `view.get_template_name(view.action)`.